### PR TITLE
Display dappUrl instead of url when the live app is dapp

### PR DESCRIPTION
**Pull request by JunichiSugiura**
### 📝 Description

Display dappUrl instead of url when the live app is dapp.

### ❓ Context

- **Impacted projects**: `ledger-live-desktop` 
- **Linked resource(s)**: `` 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 📸 Demo

Check POAP url for example.

Screenshot 2023-06-16 at 16 36 56

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts
Comment (gre) [inline apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts:6]: is there a way to avoid the introduction of `ts-expect-error`?

**Review by gre — DISMISSED** [review]

File: apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts
Comment (JunichiSugiura) [reply apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts:6]: Yes, if all union types implements type guards for each pattern.

Comment (ComradeAERGO) [reply apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts]: Added a typeguard to prevent this issue

**Comment by JunichiSugiura** [comment]
Seems the Library tests also failed on develop. So not related to this PR.
https://github.com/LedgerHQ/ledger-live/runs/14432379285

File: apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts
Comment (JunichiSugiura) [reply apps/ledger-live-desktop/src/renderer/screens/platform/v2/Catalog/Card/hooks.ts]: Thanks @ComradeAERGO . CI looks fine for this PR. So, we can merge as soon as we get an approval.

**Review by Justkant — APPROVED** [review]

**Pull request merged**
